ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala

```diff
@@ -199,7 +199,7 @@ private[sandbox] final class InMemoryLedger(
                 Some(commandId),
                 transactionId,
                 Some(`appId`),
-                _,
+                submissionId,
```

Comment:
I added the submission ID propagation for the in-memory sandbox classic. For the SQL one, it's just about using the append-only schema. Am I missing anything?

Reply:
@digital-asset/kv-participant, maybe you have an idea?

Reply:
> Sorry for making this PR rather huge. The scope was unknown beforehand. I also thought it will be easier to discuss if everything is in one place. Please let me know if I should think about splitting it.
> 
> @miklos-da @SamirTalwar-DA @andreaslochbihler-da, WDYT about the latest changes? I'm not entirely happy about them, just don't have a better idea.

If I understand correctly, after #10826 all tests (except one suite) will be running with the append-only schema and that schema will be supported from the SDK 1.18 release. @rautenrieth-da Can you confirm the latter, please?
The above implies that we should be able to remove the flag for whether submission ID is being supported (i.e., as we won't support the mutating schema anyway).

Reply:
The append-only schema is considered "production ready" in SDK 1.17, where it was described in the release notes. Hopefully, we'll be able to make it the only existing schema in SDK 1.18 (by deleting the code for the old mutating schema from our codebase).

In the meantime, we want to switch conformance tests to use the append-only schema by default. Some conformance tests will still run on the old (mutating) schema, but it's acceptable if not all optional tests are run on it. #10826 was my first attempt at modifying the conformance tests.

Reply:
Thanks for the confirmation, @rautenrieth-da.
I'd then suggest to proceed as follows:
* Make submission ID propagation the only option (i.e., this allows removing the flags introduced for choosing between `SubmissionIdPropagation.Supported` vs. `SubmissionIdPropagation.NotSupported`).
* Disable tests (if needed) for the mutating schema.
* Keep refactoring of the `CommandUpdaterFlow`, etc.